File: jarboot-ui2/src/services/PrivilegeService.ts
import Request from '@/common/Request';
import type { Privilege } from '@/types';

const urlBase = '/api/jarboot/privilege';

/**
 * 权限管理操作
 */
export default class PrivilegeService {
  public static savePrivilege(role: string, authCode: string, permission: boolean) {
    const form: FormData = new FormData();
    form.append('role', role);
    form.append('authCode', authCode);
    form.append('permission', '' + permission);
    return Request.put(urlBase, form);
  }

  public static hasPrivilege(role: string, username: string) {
    return Request.get<boolean>(urlBase, { role, username });
  }

  public static getPrivilegeByRole(role: string) {
    return Request.get<Privilege[]>(`${urlBase}/getPrivilegeByRole`, { role });
  }

  public static getPermissionInfos() {
    return Request.get(`${urlBase}/getPermissionInfos`, {});
  }
}

File: jarboot-server/src/main/java/io/github/majianzheng/jarboot/service/PrivilegeService.java
package io.github.majianzheng.jarboot.service;

import io.github.majianzheng.jarboot.entity.Privilege;

import java.util.List;

/**
 * 权限服务
 * @author majianzheng
 */
public interface PrivilegeService {
    /**
     * 获取角色的权限列表
     * @param role 角色
     * @return 权限列表
     */
    List<Privilege> getPrivilegeByRole(String role);

    /**
     * 保持权限
     * @param role 角色
     * @param authCode 权限
     * @param permission 是否拥有权限
     */
    void savePrivilege(String role, String authCode, Boolean permission);

    /**
     * 判断角色是否拥有权限
     * @param role 角色
     * @param authCode 权限
     * @return 是否有权限
     */
    boolean hasPrivilege(String role, String authCode);
}
